tensorflow/core/kernels/data/repeat_dataset_op_test.cc

```diff
@@ -0,0 +1,367 @@
+      {},
+      /*expected_output_dtypes*/ {DT_INT64, DT_STRING},
+      /*expected_output_shapes*/
+      {PartialTensorShape({2}), PartialTensorShape({1})},
+      /*expected_cardinality*/ 0,
+      /*breakpoints*/ {0, 1, 3}};
+}
+
+// Test case 3: infinite repetition.
+TestParam TestCase3() {
+  return {/*input_tensors*/
+          {DatasetOpsTestBase::CreateTensor<int64>(TensorShape{2, 1}, {1, 2})},
+          /*count*/ -1,
+          /*expected_outputs*/
+          // Use the first group of the repeated tensors to represent the
+          // infinite outputs.
+          {DatasetOpsTestBase::CreateTensor<int64>(TensorShape{1}, {1}),
+           DatasetOpsTestBase::CreateTensor<int64>(TensorShape{1}, {2})},
+          /*expected_output_dtypes*/ {DT_INT64},
+          /*expected_output_shapes*/ {PartialTensorShape({1})},
+          /*expected_cardinality*/ -1,
+          /*breakpoints*/ {0, 1, 3}};
+}
+
+class RepeatDatasetOpTestHelper : public RepeatDatasetOpTest {
+ public:
+  ~RepeatDatasetOpTestHelper() override {
+    if (dataset_) dataset_->Unref();
+  }
+
+ protected:
+  // Creates `TensorSliceDataset` variant tensor from the input vector of
+  // tensors.
+  Status CreateTensorSliceDatasetTensor(
+      std::vector<Tensor> *const tensor_vector, Tensor *dataset_tensor) {
+    DatasetBase *tensor_slice_dataset;
+    TF_RETURN_IF_ERROR(CreateTensorSliceDataset(
+        "tensor_slice_node", tensor_vector, &tensor_slice_dataset));
+    TF_RETURN_IF_ERROR(
+        StoreDatasetInVariantTensor(tensor_slice_dataset, dataset_tensor));
+    return Status::OK();
+  }
+
+  Status CreateDatasetFromTestCase(const TestParam &test_case) {
+    Tensor tensor_slice_dataset_tensor(DT_VARIANT, TensorShape({}));
+    std::vector<Tensor> input_tensors = test_case.input_tensors;
+    TF_RETURN_IF_ERROR(CreateTensorSliceDatasetTensor(
+        &input_tensors, &tensor_slice_dataset_tensor));
+    gtl::InlinedVector<TensorValue, 4> inputs;
+    Tensor count = CreateTensor<int64>(TensorShape{}, {test_case.count});
+    inputs.emplace_back(&tensor_slice_dataset_tensor);
+    inputs.emplace_back(&count);
+    TF_RETURN_IF_ERROR(CreateRepeatDatasetKernel(
+        test_case.expected_output_dtypes, test_case.expected_output_shapes,
+        &dataset_kernel_));
+    TF_RETURN_IF_ERROR(CreateRepeatDatasetContext(
+        dataset_kernel_.get(), &inputs, &dataset_kernel_ctx_));
+    TF_RETURN_IF_ERROR(CreateDataset(dataset_kernel_.get(),
+                                     dataset_kernel_ctx_.get(), &dataset_));
+    return Status::OK();
+  }
+
+  Status CreateIteratorFromTestCase(const TestParam &test_case) {
+    TF_RETURN_IF_ERROR(CreateDatasetFromTestCase(test_case));
+    TF_RETURN_IF_ERROR(
+        CreateIteratorContext(dataset_kernel_ctx_.get(), &iterator_ctx_));
+    TF_RETURN_IF_ERROR(
+        dataset_->MakeIterator(iterator_ctx_.get(), "Iterator", &iterator_));
+    return Status::OK();
+  }
+
+  std::unique_ptr<OpKernel> dataset_kernel_;
```

Comment:
Please avoid storing this state in member variables. There is no need to do this and it hurts readability.